### PR TITLE
Add benchmark output and GitHub action for monitoring.

### LINKED ISSUE #6: Benchmarks and continuous benchmarks
We want to ensure that as the project grows, memory usage and build times are kept in check.

We should investigate ways to benchmark these (i.e. by creating a benchmark project repository or generator, that can be used to benchmark builds) and add continuous benchmarking to a GitHub Action.

Relevant:  
https://github.com/rhysd/github-action-benchmark

### PR DESCRIPTION
**Pull request by Zoramite**
Fixes #6

### CODE REVIEW

File: src/profile.ts [hunk 1]
```diff
@@ -43,6 +43,19 @@ export class Profiler {
     return minBegin;
   }
 
+  benchmarkOutput(): string {
```
Comment (jeremydw) [inline src/profile.ts:46]: Nit: Naming this... `getBenchmarkOutput()` ? Reads better that way. :)

**Comment by jeremydw** [comment]
Nice! LGTM other than my naming nit. :)

File: src/profile.ts (hunk 1 above)
Comment (Zoramite) [reply src/profile.ts]: I've changed it into a getter: `get benchmarkOutput`

**Comment by Zoramite** [comment]
Benchmark data will be published to https://blinkkcode.github.io/amagaki/dev/bench/

**Comment by jeremydw** [comment]
> Benchmark data will be published to https://blinkkcode.github.io/amagaki/dev/bench/

How about just https://blinkkcode.github.io/amagaki/benchmark/ ? I kind of feel like this would be one of those things that just gets unnecessarily inserted into "dev" and then it's the only "dev" thing for years. ;)

**Comment by Zoramite** [comment]
I agree, the default path isn't very friendly. Updated it to server at `benchmark/` instead.

**Pull request merged**